PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.hdinsight/src/com/microsoft/azuretools/hdinsight/projects/HDInsightScalaPageTwo.java

```diff
@@ -55,6 +58,40 @@ public void configureJavaProject(String newProjectCompliance, IProgressMonitor m
 			if (parent == null) { 
 				parent = (HDInsightsScalaProjectWizard) this.getWizard();
 			}
+			
+			if (hasConfiguredScalaClasspathContainer == false) {
+				hasConfiguredScalaClasspathContainer = true;
+				Display.getDefault().syncExec(new Runnable() {
+				    public void run() {
+						IJavaProject javaProject = getJavaProject();
+						IClasspathEntry[] entries = null;
+						try {
+							entries = javaProject.getRawClasspath();
+							IClasspathEntry scalaClasspathContainerEntry = null;
+							int scalaClasspathContainerEntryIndex = -1;
+							for (int i = 0; i < entries.length; i++) {
+								String entryName = entries[i].getPath().toPortableString().toLowerCase();
+								if (entryName != null && entryName.contains(parent.scalaClasspathContainerId.toLowerCase())) {
+									scalaClasspathContainerEntry = entries[i];
+									scalaClasspathContainerEntryIndex = i;
+									break;
+								}
+							}
+							
+							if (scalaClasspathContainerEntry != null) {
+								IClasspathEntry created = BuildPathDialogAccess.configureContainerEntry(getShell(), scalaClasspathContainerEntry, javaProject, entries);
+								if (created != null) {
+									entries[scalaClasspathContainerEntryIndex] = created;
+								}
+							}
+							
```

Comment:
How about the failure path of those various being `null`?

Reply:
Good point. If it is null, should not setRawClasspath since there is no change.

---
In reply to: [144226962](https://github.com/Microsoft/azure-tools-for-java/pull/1098#discussion_r144226962) [](ancestors = 144226962)